org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/validation/AddMarkersOperation.java

```diff
@@ -108,4 +138,20 @@ protected void execute(final IProgressMonitor monitor) throws CoreException, Inv
 		}
 	}
 
+	private void deleteAnnotations() {
+		if (annotationModel != null) {
+			@SuppressWarnings("unchecked")
+			Iterator<Annotation> iterator = annotationModel.getAnnotationIterator();
+			List<Annotation> toRemove = new ArrayList<>();
+			while (iterator.hasNext()) {
+				Annotation next = iterator.next();
```

Comment:
We should only remove annotations that are not yet marked as deleted

Reply:
Good point, I will add " && !next.isMarkedDeleted()"